.github/workflows/website-deploy-preview.yml

```diff
@@ -0,0 +1,69 @@
+name: "website-deploy-preview"
+
+on:
+  workflow_dispatch:
+
+  pull_request:
+    types:
+      - opened
+      - synchronize
+      - reopened
+    paths:
+      - "website/**"
+
+env:
+  AWS_REGION: us-east-2
+  IAM_ROLE_ARN: arn:aws:iam::068007702576:role/cplive-plat-ue2-dev-atmos-docs-github-action
```

Comment:
How does this work to login to awe role with ubuntu-latest runner instead of self-hosted?

Reply:
to use the GitHub hosted runners, we need to change the scope for it from `cloudposse/infra-live` to `cloudposse` allowing all our repos to use them.
I'm not sure we want to do it or discussed that.

Using `ubuntu-latest` the or the two reasons:

It works on ubuntu-latest runner instead of self-hosted b/c we provision a GitHub OIDC provider in each account and an IAM role that the provider is assuming. It's all in one account so no cross-account permissions are required.

We are not using the hosted GitHub runners instance role/profile because the OIDC provider seems to allow better control (we create a separate IAM role for each app/service, instead of putting all the permissions (and the future ones) into the instance profile roles of the hosted GitHub runners.
Also, it is cross-account, so we will have to add the required permissions on both sides, which could be challenging to do on the GitHub runners side b/c we'll have to add a lot of diff permissions in one place.

In short, the OIDC provider with the IAM role per repo/app works exactly like the EKS OIDC provider for Kubernetes service accounts - it allow creating small roles with just the required permissions, and not having cross-account permissions

Reply:
the OIDC provider can also we be with self-hosted runners (w/o any changes to the provider or to the iAM role) - we just need to change the GitHub scope from `cloudposse/infra-live` to `cloudposse` in the `github-runners` component (if we want to do that)